### PR TITLE
Improve stops view

### PR DESCRIPTION
**Pull request by abrahammartin**
I have removed the old stops view which did a call to retrieve all bus stops (too long and timed out) and substituted that for one that is restricted to Cambridge by default or by Transport Area models (which we can define in admin). These are not drawn in a map and the bus stops are clickable, instead of showing just a searchable list.

It also includes an important speed improvement for the Stops API in https://github.com/SmartCambridge/tfc_web/pull/270/commits/de6d93f5974915a1a97df23215d7cd0dfbaed660

### CODE REVIEW

File: tfc_web/transport/templates/transport/bus_stops_map.html [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+{% block head %}
+    {% include 'leaflet_snippet.html' %}
+{% endblock %}
+{% block content %}
+    <div style="height: 80vh;">
+        <div id="title" style="height: 5%; width: 100%;"></div>
+        <div id="map" style="height: 90%; width: 100%;"></div>
+    </div>
+    <script>
+        var map = L.map('map').setMinZoom(15);
```
Comment (jw35) [inline tfc_web/transport/templates/transport/bus_stops_map.html]: With zoom limited to 15 or above I find navigating anywhere difficult. If the problem is having too many stop icons when zoomed out then IMO better is either to suppress them below some zoom level or to use something like https://github.com/Leaflet/Leaflet.markercluster

File: tfc_web/transport/templates/transport/bus_stops_map.html [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        $(document).ready(function() {
+            L.tileLayer('http://{s}.tile.osm.org/{z}/{x}/{y}.png', {
+                attribution: '&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
+            }).addTo(map);
+            map.fitBounds([
+                {% for a in area %}
+                    [{{ a.1 }}, {{ a.0 }}],
+                {% endfor %}
+            ]);
+            map.setMaxBounds([
```
Comment (jw35) [inline tfc_web/transport/templates/transport/bus_stops_map.html]: For me, this results in a largely empty map centred on the Barton Rifle range, which isn't all that useful...

**Comment by jw35** [comment]
Is this view actually linked from anywhere - I know the spiders find it occasionally, but I can't?

Vaguely related, what is https://tfc-app4.cl.cam.ac.uk/transport/areas/ suppose to do?

**Comment by jw35** [comment]
Closes #266

File: tfc_web/transport/templates/transport/bus_stops_map.html (hunk 1 above)
Comment (abrahammartin) [reply tfc_web/transport/templates/transport/bus_stops_map.html]: Done in https://github.com/SmartCambridge/BusStopChooser/pull/6 and then backported here in further commits

File: tfc_web/transport/templates/transport/bus_stops_map.html (hunk 2 above)
Comment (abrahammartin) [reply tfc_web/transport/templates/transport/bus_stops_map.html]: redone completely now using BusStopChooser

**Comment by abrahammartin** [comment]
Ready for review again

**Comment by abrahammartin** [comment]
Changes from https://github.com/SmartCambridge/BusStopChooser/pull/6 implemented now. One thing that has occurred to me is that this web view won't work unless the user has logged in and had accepted the API terms and conditions before. So I guess we want to make this website protected and redirect the user to sign up and accept the t&cs?

**Comment by jw35** [comment]
Argh! You can no longer actually select stops when the chooser is used with its 'internal' onclick handler in the SmartPanel or Pocket SmartPanel config pages!

**Comment by jw35** [comment]
> But it's not that simple, because it looks as if leaflet.markercluster needs to be loaded after Leaflet and as things stand I think Leaflet gets loaded from stop_bus_map's requirements _after_ stop_timetable's requirements and there's no way to control this.

Worth noting that @ijl20 is rewriting the widget requirements code to address this.

**Comment by jw35** [comment]
> One thing that has occurred to me is that this web view won't work unless the user has logged in and had accepted the API terms and conditions before.

No, it should work anonymously. We pass our own internal API token to BusStopChooser to authorise it's API calls. Is there any other problem?

**Comment by abrahammartin** [comment]
> > One thing that has occurred to me is that this web view won't work unless the user has logged in and had accepted the API terms and conditions before.
> 
> No, it should work anonymously. We pass our own internal API token to BusStopChooser to authorise it's API calls. Is there any other problem?

Ah! ok, I missed that

**Pull request merged**